HashTable/HashTable.c

```diff
@@ -0,0 +1,208 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include "HashTable.h"
+
+typedef struct Node {
+    char* word;
+    int count;
+    struct Node* next;
+} Node;
+
+typedef struct HashTable {
+    Node** segments;
+    int countOfSegments;
+    int countOfElements;
+} HashTable;
+
+HashTable* createTable(int size) {
+    HashTable* table = malloc(sizeof(HashTable));
+    if (table == NULL) {
+        return NULL;
+    }
+    table->countOfSegments = size;
+    table->countOfElements = 0;
+    table->segments = malloc(size * sizeof(Node*));
+    if (table->segments == NULL) {
+        return NULL;
+    }
+    for (int i = 0; i < size; i++) {
+        table->segments[i] = NULL;
+    }
+    return table;
+}
+
+unsigned int hashFunction(char* word, int size) {
+    unsigned int hash = 0;
+    for (int i = 0; word[i] != '\0'; i++) {
+        hash = (hash * 31 + word[i]) % size;
+    }
+    return hash;
+}
+
+void resizeHashTable(HashTable* table) {
+    int oldSegmentCount = table->countOfSegments;
+    Node** oldSegments = table->segments;
+
+    int newSegmentCount = oldSegmentCount * 2;
+    table->segments = calloc(newSegmentCount, sizeof(Node*));
+    table->countOfSegments = newSegmentCount;
+    table->countOfElements = 0;
+
+    for (int i = 0; i < oldSegmentCount; ++i) {
+        Node* current = oldSegments[i];
+        while (current) {
+            addWord(table, current->word);
+            Node* temp = current;
+            current = current->next;
+            free(temp->word);
+            free(temp);
+        }
+    }
+
+    free(oldSegments);
+}
+
+void addWord(HashTable* table, char* word) {
+    int indexOfWord = hashFunction(word, table->countOfSegments);
```

Comment:
```suggestion
    unsigned int indexOfWord = hashFunction(word, table->countOfSegments);
```
Иначе большое unsigned значение откастается к отрицательному числу